### PR TITLE
Add new `static` prop for SSR rendering

### PR DESCRIPTION
**Pull request by damassi**
This PR adds a new `static` config prop, which when true will skip rendering of a React portal, and simply render children as they are passed into the carousel. This smooths the handshake between client / server rendering, however with the tradeoff: when `static`, the carousel cannot be modified at runtime (e.g., slides cannot be dynamically added or removed). For many usecases this tradeoff is acceptable. See [this comment](https://github.com/theolampert/react-flickity-component/pull/64#issuecomment-490527221) for more info. 

#### Description of problem

While attempting to implement the carousel component, we noticed that there was a flicker when attaching to the DOM from an SSR pass: 

flicker

Initially we thought that it was an issue with the underlying flickity instance on init, but digging deeper found that the problem came down to this bit of code, in `react-flickity-component`: 

```js
renderPortal() {	
  if (!this.carousel) return null;
  ...
```

During the initialization phase, the carousel instance hasn't yet been instantiated, and so `null` is returned from within react, leading to empty content on the page. When things init, the code proceeds and things execute correctly, albeit with a flicker. 

However, when playing more with the implementation we found that the portal is unnecessary, even if the carousel is **only** rendered client-side. By simply returning `this.props.children` in the `render` method

```js
  render() {
    return React.createElement(
      this.props.elementType,
      {
        className: this.props.className,
        ref: c => {
          this.carousel = c;
        },
      },
      this.props.children
    );
  }
```

everything attaches and initializes as it should, with the added benefit that it also works correctly during the server -> client handshake pass. 

@theolampert - Wondering your thoughts? Why was `renderPortal` used to begin with? By removing it everything seems to work a-ok: 

#### No flicker: 

no-flicker

We've only been able to test this for our use-case but yeah! Any feedback would be appreciated. 

(If you're curious about our SSR implementation, see [here](https://github.com/artsy/reaction/blob/master/src/Components/v2/CarouselV3.tsx#L160-L171).)

### CODE REVIEW

**Comment by theolampert** [comment]
@damassi Thanks for the details PR, I need to take a closer look over this. @yaodingyd could you shed some light on the `renderPortal` method?

**Comment by yaodingyd** [comment]
Thanks for your contribution. Unfortunately we do have to use portal to render children. Your code works perfectly fine if your slider items are static, meaning no adding or deleting items. But if they are dynamic, it would break the component because after Flickity is initialized, the children are not rendered in the original DOM hierarchy and we have to use Portal to render them correctly.

**Comment by damassi** [comment]
@theolampert / @yaodingyd - would y'all be open to adding a `static` configuration prop to toggle this behavior on and off?

**Comment by yaodingyd** [comment]
sounds good to me

**Comment by damassi** [comment]
Cool, i'll update this PR today.

File: src/index.js [hunk 1]
```diff
@@ -65,27 +65,29 @@ class FlickityComponent extends Component {
...
-  options: PropTypes.object,
+  children: PropTypes.array,
   className: PropTypes.string,
+  disableImagesLoaded: PropTypes.bool,
   elementType: PropTypes.string,
-  children: PropTypes.array,
   flickityRef: PropTypes.func,
+  options: PropTypes.object,
+  reloadOnUpdate: PropTypes.bool,
+  static: PropTypes.bool,
```
Comment (damassi) [inline src/index.js]: Just alpha

**Comment by damassi** [comment]
Alright @theolampert - added a commit with `static` and updated PR description / title. Let me know if you need anything else.

**Comment by yaodingyd** [comment]
@theolampert  I think I'll update readme later with the discussion of this PR.

**Comment by damassi** [comment]
Hi @theolampert - would you mind publishing this new version to NPM? Noticed the latest is still `3.2.0`.

**Comment by theolampert** [comment]
@damassi done! please check `3.3.0`

**Pull request merged**